app/code/Magento/Sales/Test/Mftf/Test/StorefrontPlaceOrderAndCheckAddressesAreDifferentOnAdminOrderView.xml

```diff
@@ -0,0 +1,98 @@
+    <test name="StorefrontPlaceOrderAndCheckAddressesAreDifferentOnAdminOrderView">
+        <annotations>
+            <stories value="Github Issue: #31608 Shipping address region leaking into billing address"/>
+            <title value="Billing and Shipping addresses should show correct data on Admin Order View"/>
+            <description value="Place order on Store Front with manually filled billing address state and selected shipping address state. Check that billing address show correct state on Admin Order View page"/>
+            <severity value="MINOR"/>
+            <group value="sales"/>
+            <group value="GI31608"/>
+        </annotations>
+        <before>
+            <!--Create simple product.-->
+            <createData entity="SimpleProduct2" stepKey="createSimpleProduct"/>
+
+            <!-- Create Customer Account -->
+            <createData entity="Customer_UK_US" stepKey="createCustomer"/>
+
+            <comment userInput="Adding the comment to replace 'indexer:reindex' command for preserving Backward Compatibility" stepKey="reindex"/>
+            <comment userInput="Adding the comment to replace 'cache:flush' command for preserving Backward Compatibility" stepKey="flushCache"/>
+        </before>
+        <after>
+            <deleteData createDataKey="createCustomer" stepKey="deleteCreateCustomer"/>
+            <deleteData createDataKey="createSimpleProduct" stepKey="deleteSimpleProduct"/>
+
+            <!-- Admin logout -->
+            <actionGroup ref="AdminOrdersGridClearFiltersActionGroup" stepKey="clearFilters"/>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+
+            <!-- Customer Logout-->
+            <actionGroup ref="StorefrontCustomerLogoutActionGroup" stepKey="logoutCustomer"/>
+
+            <!-- Reindex invalidated indices after product attribute has been created/deleted -->
+            <magentoCron groups="index" stepKey="reindexInvalidatedIndices"/>
+        </after>
+
+        <!-- Order a product -->
+        <amOnPage url="{{StorefrontProductPage.url($$createSimpleProduct.custom_attributes[url_key]$$)}}" stepKey="navigateToProductPage"/>
+        <waitForPageLoad stepKey="waitForSimpleProductPageLoad"/>
+        <actionGroup ref="StorefrontAddProductToCartActionGroup" stepKey="cartAddSimpleProductToCart">
+            <argument name="product" value="$$createSimpleProduct$$"/>
+            <argument name="productCount" value="1"/>
+        </actionGroup>
+
+        <!-- Got to checkout -->
+        <actionGroup ref="GoToCheckoutFromMinicartActionGroup" stepKey="navigateToCheckout"/>
+        <waitForPageLoad stepKey="waitForPaymentSelectionPageLoad"/>
+
+        <!-- Login as customer on checkout page -->
+        <actionGroup ref="LoginAsCustomerOnCheckoutPageActionGroup" stepKey="loginAsCustomer">
+            <argument name="customer" value="$$createCustomer$$"/>
+        </actionGroup>
```

Comment:
I think logging in as a customer is not necessary, as the bug is also reproducible for guest customers as well. So let's focus our test for guest users. 
ℹ️ But if you confirm that the bug is also reproducible for logged in customers, then you may skip all my related comments to customer and addresses.
What do you think?

Could you please adjust this part in order to fill the shipping address fields?
You may try to check if any of these Action Groups are suitable here:
- FillGuestCheckoutShippingAddressWithCountryActionGroup
- FillShippingAddressOneStreetActionGroup

```suggestion
```